soapui/src/main/java/com/eviware/soapui/impl/rest/actions/oauth/OAuth2TokenExtractor.java

```diff
@@ -192,6 +192,12 @@ void refreshAccessToken(OAuth2Parameters parameters) throws OAuthProblemExceptio
         OAuthToken oAuthToken = oAuthClient.accessToken(accessTokenRequest, OAuthJSONAccessTokenResponse.class).getOAuthToken();
         parameters.applyRetrievedAccessToken(oAuthToken.getAccessToken());
         parameters.setAccessTokenIssuedTimeInProfile(TimeUtils.getCurrentTimeInSeconds());
+	// Due to RFC6749 6. the client MUST discard the old refresh token if the server issues a new token
+	// and replace it with the new one.
+	String newRefreshToken = oAuthToken.getRefreshToken();
+	if (newRefreshToken!=null && newRefreshToken!=parameters.refreshToken) {
```

Comment:
`newRefreshToken!=parameters.refreshToken` looks always `true` so redundant: this operator compares references to the strings but not content.